disdrodb/data_transfer/upload_data.py

```diff
@@ -0,0 +1,261 @@
+
+    filtered = []
+
+    for metadata_fpath in metadata_fpaths:
+        metadata_dict = _read_yaml_file(metadata_fpath)
+        if metadata_dict.get("data_url"):
+            print(f"{metadata_fpath} already has a remote url specified. Skipping.")
+            continue
+        filtered.append(metadata_fpath)
+
+    return filtered
+
+
+def _upload_data_to_zenodo(
+    metadata_fpaths: List[str], files_compression: Optional[str] = None, sandbox: bool = False
+) -> None:
+    """Upload data to Zenodo.
+
+    Parameters
+    ----------
+    metadata_fpaths: list of str
+        List of metadata file paths.
+    sandbox: bool
+        If True, upload to Zenodo sandbox for testing purposes.
+    """
+
+    deposition_id, bucket_url = _create_zenodo_deposition(sandbox)
+    zenodo_host = "sandbox.zenodo.org" if sandbox else "zenodo.org"
+    deposition_url = f"https://{zenodo_host}/deposit/{deposition_id}"
+    print(f"Zenodo deposition created: {deposition_url}.")
+
+    for metadata_fpath in metadata_fpaths:
+        remote_path = _upload_station_data_to_zenodo(metadata_fpath, bucket_url, files_compression)
+        _update_metadata_with_zenodo_url(metadata_fpath, deposition_id, remote_path, sandbox)
+
+    print("Data uploaded. Please review your deposition an publish it when ready.")
+
+
+def _generate_data_remote_path(metadata_fpath: str) -> str:
+    """Generate data remote path from a metadata path.
+
+    metadata_fpath has the form "disdrodb_dir/Raw/data_source/campaign_name/metadata/station_name.yml".
+    The remote path has the form "data_source/campaign_name/station_name".
+
+    Parameters
+    ----------
+    metadata_fpath: str
+        Metadata file path.
+    """
+
+    remote_path = os.path.normpath(metadata_fpath)
+    # Remove up to "Raw/"
+    remote_path = remote_path.split("Raw" + os.sep)[1]
+    # Remove "/metadata"
+    remote_path = remote_path.replace(os.sep + "metadata", "")
+    # Remove trailing ".yml"
+    remote_path = os.path.splitext(remote_path)[0]
+
+    return remote_path
+
+
+def _upload_station_data_to_zenodo(metadata_fpath: str, bucket_url: str, files_compression: Optional[str]) -> str:
+    """Zip and upload station data to Zenodo.
+
+    Update the metadata file with the remote url, and zip the data directory before uploading.
+
+    Parameters
+    ----------
+    metadata_fpath: str
+        Metadata file path.
+    bucket_url: str
+        Zenodo bucket url.
+    files_compression: str
+        Compression method for individual files inside station data.
+    """
+
+    remote_path = _generate_data_remote_path(metadata_fpath)
+    remote_url = f"{bucket_url}/{remote_path}.zip"
+    temp_zip_path = _archive_station_data(metadata_fpath, files_compression)
+
+    _upload_file_to_zenodo(temp_zip_path, remote_url)
+
+    os.remove(temp_zip_path)
+
+    return remote_path
+
+
+def _archive_station_data(metadata_fpath: str, files_compression: Optional[str] = None) -> str:
```

Comment:
The download functions expect the station directory to be a zip directory. So I would not provide the option here to choose the compression method, but fix it to `zip` (unless you want to adapt the download code to decompress according to the file extension).

Reply:
ok, goes with next command